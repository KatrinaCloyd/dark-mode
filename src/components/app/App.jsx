/* eslint-disable indent */
import React from 'react';
import CharacterList from './CharacterList';
import Header from './Header';
import './app.css';

export default function App() {
  return (
    <div >
      <Header />
      <CharacterList />
    </div>
  );
}
