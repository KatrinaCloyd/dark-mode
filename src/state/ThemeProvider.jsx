/* eslint-disable indent */
import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

export const ThemeContext = createContext();

export const ThemeProvider = ({ children }) => {
    const [party, setParty] = useState(false);

    const partyOn = () => setParty((party) => !party);

    return (
        <ThemeContext.Provider value={{ party, partyOn }}>
            {children}
        </ThemeContext.Provider>
    );
};

ThemeProvider.propTypes = {
    children: PropTypes.node.isRequired
};
