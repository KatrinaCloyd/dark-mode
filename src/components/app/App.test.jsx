/* eslint-disable max-len */
/* eslint-disable indent */
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import App from './App';
import { ThemeProvider } from '../../state/ThemeProvider';

const server = setupServer(
    rest.get(
        'https://www.breakingbadapi.com/api/characters',
        (req, res, ctx) => {
            return res(ctx.json([
                {
                    'char_id': 1,
                    'name': 'Walter White',
                    'birthday': '09-07-1958',
                    'occupation': [
                        'High School Chemistry Teacher',
                        'Meth King Pin'
                    ],
                    'img': 'https://images.amcnetworks.com/amc.com/wp-content/uploads/2015/04/cast_bb_700x1000_walter-white-lg.jpg',
                    'status': 'Presumed dead',
                    'nickname': 'Heisenberg',
                    'appearance': [
                        1,
                        2,
                        3,
                        4,
                        5
                    ],
                    'portrayed': 'Bryan Cranston',
                    'category': 'Breaking Bad',
                    'better_call_saul_appearance': []
                },
                {
                    'char_id': 2,
                    'name': 'Jesse Pinkman',
                    'birthday': '09-24-1984',
                    'occupation': [
                        'Meth Dealer'
                    ],
                    'img': 'https://vignette.wikia.nocookie.net/breakingbad/images/9/95/JesseS5.jpg/revision/latest?cb=20120620012441',
                    'status': 'Alive',
                    'nickname': 'Cap n\' Cook',
                    'appearance': [
                        1,
                        2,
                        3,
                        4,
                        5
                    ],
                    'portrayed': 'Aaron Paul',
                    'category': 'Breaking Bad',
                    'better_call_saul_appearance': []
                },
                {
                    'char_id': 3,
                    'name': 'Skyler White',
                    'birthday': '08-11-1970',
                    'occupation': [
                        'House wife',
                        'Book Keeper',
                        'Car Wash Manager',
                        'Taxi Dispatcher'
                    ],
                    'img': 'https://s-i.huffpost.com/gen/1317262/images/o-ANNA-GUNN-facebook.jpg',
                    'status': 'Alive',
                    'nickname': 'Sky',
                    'appearance': [
                        1,
                        2,
                        3,
                        4,
                        5
                    ],
                    'portrayed': 'Anna Gunn',
                    'category': 'Breaking Bad',
                    'better_call_saul_appearance': []
                }]));
        }
    )
);

describe('Breaking bad display mode page', () => {
    beforeAll(() => server.listen());
    afterAll(() => server.close());

    it('displays a list of breaking bad characters with styling based on checkbox', async () => {
        const homePage = render(
            <ThemeProvider >
                <App />
            </ThemeProvider >);
        expect(homePage).toMatchSnapshot();

        const hdr = screen.getByLabelText('hdr');
        expect(hdr).toHaveClass('headLight');

        const toggle = await screen.findByRole('checkbox');
        expect(toggle).toBeVisible();

        //test for the <ul> that contains the BB characters:
        const charList = await screen.findAllByRole('list', {
            name: 'characters',
        });
        expect(charList).toHaveLength(3);

        fireEvent.click(toggle);
        expect(hdr).toHaveClass('headParty');

    });
});
