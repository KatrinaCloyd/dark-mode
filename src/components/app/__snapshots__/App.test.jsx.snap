// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breaking bad display mode page displays a list of breaking bad characters with styling based on checkbox 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          aria-label="hdr"
          class="headLight"
        >
          <h1>
            Breaking Bad Character Party... 
            <br />
            or not
          </h1>
          <div
            class="controls"
          >
            <h4>
              boring
            </h4>
            <label
              class="switch"
            >
              <input
                type="checkbox"
              />
              <span
                class="slider"
              />
            </label>
            <h4>
              partay
            </h4>
          </div>
        </div>
        <img
          src="https://cdn.dribbble.com/users/841405/screenshots/2541549/animaion-bb.gif"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        aria-label="hdr"
        class="headLight"
      >
        <h1>
          Breaking Bad Character Party... 
          <br />
          or not
        </h1>
        <div
          class="controls"
        >
          <h4>
            boring
          </h4>
          <label
            class="switch"
          >
            <input
              type="checkbox"
            />
            <span
              class="slider"
            />
          </label>
          <h4>
            partay
          </h4>
        </div>
      </div>
      <img
        src="https://cdn.dribbble.com/users/841405/screenshots/2541549/animaion-bb.gif"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
